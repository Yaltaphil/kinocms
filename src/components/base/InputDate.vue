<template>
    <input
        ref="input"
        type="date"
        :value="dateToYYYYMMDD(value)"
        @input="updateValue($event.target)"
        @focus="selectAll"
    />
</template>

<script>
export default {
    props: {
        value: {
            type: Date,
            default: () => new Date(),
        },
    },
    methods: {
        dateToYYYYMMDD(d) {
            // alternative implementations in https://stackoverflow.com/q/23593052/1850609
            return (
                d &&
                new Date(d.getTime() - d.getTimezoneOffset() * 60 * 1000)
                    .toISOString()
                    .split("T")[0]
            );
        },
        updateValue: function (target) {
            this.$emit("input", target.valueAsDate);
        },
        selectAll: function (event) {
            // Workaround for Safari bug
            // http://stackoverflow.com/questions/1269722/selecting-text-on-focus-using-jquery-not-working-in-safari-and-chrome
            setTimeout(function () {
                event.target.select();
            }, 0);
        },
    },
};
</script>
