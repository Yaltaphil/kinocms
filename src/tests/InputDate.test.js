import { mount } from "@vue/test-utils";
import InputDate from "@/components/base/BaseSwitcher.vue";

describe("Testing InputDate Component", () => {
    const someDate = new Date(1341546546);
    const wrapper = mount(InputDate, {
        propsData: { value: someDate },
    });
    test("This dater exists and visible:", () => {
        expect(wrapper.exists()).toBeTruthy();
        expect(wrapper.isVisible()).toBeTruthy();
    });

    // test("Props comes to it:", () => {
    //     expect(wrapper.props("value")).toBeDefined();
    // });

    test("InputDate matches snapshot:", () => {
        expect(wrapper.element).toMatchSnapshot();
    });
});
