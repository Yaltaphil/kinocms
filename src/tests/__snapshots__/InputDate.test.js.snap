// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing InputDate Component InputDate matches snapshot: 1`] = `
<div
  class="form-group"
  value="Fri Jan 16 1970 15:39:06 GMT+0300 (Москва, стандартное время)"
>
  <div
    class="
            custom-control
            custom-switch
            custom-switch-off-danger
            custom-switch-on-success
        "
  >
    <input
      class="custom-control-input"
      id="1"
      type="checkbox"
    />
     
    <label
      class="custom-control-label"
      for="1"
    />
  </div>
</div>
`;
